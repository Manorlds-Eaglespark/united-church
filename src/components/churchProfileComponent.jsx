/* eslint-disable react/no-unused-state */
/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import viewAllChurchesAction from '../actions/churches/viewAllChurchesAction';
import history from '../utils/history';
import theBannerImage from '../assets/images/take_action1.png';
import churchIcon from '../assets/images/church.png';
import Header from './common/Header';

export class churchProfileComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  componentDidMount() {
    const { fetchAllChurches } = this.props;
    fetchAllChurches();
  }

    render() {
      const { churches } = this.props;
      const churchesList = churches.map(church => (
        <div className="card" key={church.id}>
          <div className="row article">
            <div className="col-md-12 col-lg-5">
              <Link to={`/${church.id}`}>
                <img className="img-article" alt="article_image" src={churchIcon} />
              </Link>
            </div>
            <div className="col">
                <h5 className="name">{church.name}</h5>
              <p className="description">
                            Pastor:
                {' '}
                {church.pastor}
              </p>
              <div className="article-details">

                <span>
                                Service times:
                  {' '}
                  {church.service_time}
                </span>
              </div>
              <div className="article-details">

                <span>
                                Location:
                  {' '}
                  {church.location}
                </span>
              </div><div className="article-details">

<span>
                Contact:
  {' '}
  {church.contact}
</span>
</div>
            </div>
          </div>
        </div>
      ));


      return (
        <div>
            <Header/>
          <div>
            <div className="main">
              <div className="head-liner">
                            &nbsp;
              </div>

              <div className="father">
                <img className="bannerImage" src={theBannerImage} alt="" />
              </div>

              <div className="head-liner">
                            &nbsp;
              </div>


              <div className="father">
                <h2>Church List</h2>
                <div className="row projects">{churchesList}</div>
              </div>


              <div className="head-liner">
                            &nbsp;
              </div>

            </div>

          </div>
          <footer/>

        </div>
      );
    }
}


churchProfileComponent.propTypes = {
  churches: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      location: PropTypes.string,
      pastor: PropTypes.string,
      service_time: PropTypes.string,
      contact: PropTypes.string
    }),
  ).isRequired,
};

churchProfileComponent.propTypes = {
    fetchAllChurches: PropTypes.func.isRequired,
};

export const mapStateToProps = (state) => {
  const { churches } = state.viewAllChurchesReducer;
  return { churches };
};

export default connect(mapStateToProps,
  { fetchAllChurches: viewAllChurchesAction })(churchProfileComponent);
