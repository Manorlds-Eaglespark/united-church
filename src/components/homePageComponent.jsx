import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import viewAllMemberPostsAction from '../actions/churches/viewAllMemberPostsAction';
import theBannerImage from '../assets/images/take_action1.png';
import Header from '../components/common/Header'
import churchIcon from '../assets/images/church.png';

export class homePageComponent extends Component {

  componentDidMount() {
    const { fetchMemberPosts } = this.props;
    fetchMemberPosts();
  }

  render() {
    const { memberPosts } = this.props;
    const allMemberPosts = memberPosts.map(memberPost => (
      <div className="post-body" key={memberPost.id}>
        <div className="row article">
          <div className="post-title-section">
            <p className="time-ago">10 minutes ago</p>
              <img className="profile-image" alt="article_image" src="http://www.newdesignfile.com/postpic/2014/07/generic-user-icon-windows_352871.png" />
              <p className="profile-text">{memberPost.name}</p>
          </div>
          <div className="post-body-section">
            <img src={memberPost.image_url} alt="" />
            <p className="post-message-text">{memberPost.message}</p>
            </div>
          </div>
        </div>
    ));


    return (
      <div>
        <Header />
        <div>
          <div className="main">
            <div className="head-liner">
              &nbsp;
              </div>

            <div className="father">
              <img className="bannerImage" src={theBannerImage} alt="" />
            </div>

            <div className="head-liner">
              &nbsp;
              </div>


            <div className="father">
              <h2 className="color-green">Testimonies, praise, worship, encouragement and prayer requests</h2>
              <div className="post-container">
              <div className="row projects">{allMemberPosts}</div>
              </div>
            </div>


            <div className="head-liner">
              &nbsp;
              </div>
          </div>
        </div>
        <footer />
      </div>
    );
  }
}

homePageComponent.propTypes = {
  memberPosts: PropTypes.arrayOf(
    PropTypes.shape({
      created_on: PropTypes.string,
      image_url: PropTypes.string,
      message: PropTypes.string,
      name: PropTypes.string,
      id: PropTypes.number,
    }),
  ).isRequired,
};

homePageComponent.propTypes = {
  fetchMemberPosts: PropTypes.func.isRequired,
};

export const mapStateToProps = (state) => {
  const { memberPosts } = state.viewAllMemberPostsReducer;
  return { memberPosts };
};

export default connect(mapStateToProps,
  { fetchMemberPosts: viewAllMemberPostsAction })(homePageComponent);
