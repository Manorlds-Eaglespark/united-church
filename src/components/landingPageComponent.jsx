import React from 'react';
import { Link } from 'react-router-dom';
import logoImage from '../assets/images/login_image.png';
import loginImage from '../assets/images/google.png';

const landingPageComponent = () => (
  <div>
    <div className="pageHeading">
      <span className="logo-r">United Church</span>
    </div>

    <div>
    <img className="center landing_image" src={logoImage} alt="" />
    <p className="logo-text color-blue">Sharing across a united CHURCH </p>
    <Link to="/select">
      <img className="center google-login" src={loginImage} alt="" />
    </Link>
  </div>
  </div>
);

export default landingPageComponent;
