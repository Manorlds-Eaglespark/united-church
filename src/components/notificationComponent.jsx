import React from 'react';
import Header from '../components/common/Header'
import footer from '../components/common/Footer'

const notificationComponent = () => (
    <div>
        <Header />
        <footer/>
    </div>
);

export default notificationComponent;
