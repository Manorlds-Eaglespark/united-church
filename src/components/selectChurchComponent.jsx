import React from 'react';
import { Link } from 'react-router-dom';
import Header from '../components/common/Header'
import continueButton from '../assets/images/continue_button_gray.jpg';
const selectChurchComponent = () => (
    <div>
        <Header />
        <div>

            <div className="father">
                <div className="brother">
                    <p className="logo-text">Sharing across a united CHURCH</p>
                </div>
                <div className="sister">
                    <div className="box">

                        <p className="color-blue">Select a Church you attend</p>

                        <select>
                            <option>Watoto</option>
                            <option>Life Church</option>
                            <option>Bukoto C/U</option>
                            <option>St. Mathias Down town</option>
                            <option>Reedemed Church</option>
                        </select>

                        <Link to="/home"><img className="center continue-button" src={continueButton} alt="Continue" /></Link>
                    </div>
                </div>
            </div>
        </div>
    </div>
);

export default selectChurchComponent;
