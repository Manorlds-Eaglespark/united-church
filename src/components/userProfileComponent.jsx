import React from 'react';
import Header from '../components/common/Header'

const userProfileComponent = () => (
    <div>
        <Header />
        <div>
            <h2 className="profile-title">Your Profile Card</h2>
            <div className="card center-user-profile">
                <img src="https://cdn.pixabay.com/photo/2016/08/08/09/17/avatar-1577909_960_720.png" alt="John" />
                <h1>John Kamogs</h1>
                <p className="title">Prayer warrior, Harvest Church</p>
                <p>Bukoto, Kampala</p>
            </div>
        </div>
        <footer />
    </div>
);

export default userProfileComponent;
