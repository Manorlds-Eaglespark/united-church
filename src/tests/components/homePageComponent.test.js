import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { homePageComponent } from '../../components/homePageComponent';

describe('homePageComponent', () => {
  let wrapper;
  const storeItems = {};
  const mockStore = configureStore();
  const store = mockStore(storeItems);

  beforeEach(() => {
    wrapper = shallow(<homePageComponent />);
    jest.resetAllMocks();
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });


  it('should call handleLogOutClick function', () => {
    const instance = wrapper.instance();
    console.log('bjnbkjn the instance', instance);
    // const spy = jest.spyOn(homePageComponent.prototype, 'handleLogOutClick');
    // const wrapper = shallow(<homePageComponent {...props} />);
    // wrapper.instance().handleLogOutClick();
    // expect(spy).toHaveBeenCalled();
  });

//   it('Testing mapStateToProps', () => {
//     const initialState = {
//       redflags: [],
//     };
//     expect(mapStateToProps(initialState)).toEqual({
//       viewAllRedFlagsReducer: {},
//     });
//   });
});
