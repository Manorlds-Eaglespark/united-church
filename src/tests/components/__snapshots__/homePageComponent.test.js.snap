// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homePageComponent matches snapshot 1`] = `<homePageComponent />`;
